### PR TITLE
Allow retrieving the final URL from the response

### LINKED ISSUE #136: Add a method on Response for retrieving the final URL
It would seem that there's no equivalent to *reqwest*'s [`url`](https://docs.rs/reqwest/latest/reqwest/blocking/struct.Response.html#method.url) method on the `Response` type?

### PR DESCRIPTION
**Pull request by baskerville**
Fixes #136.

### CODE REVIEW

File: src/parsing/response.rs [hunk 1]
```diff
@@ -85,6 +86,7 @@ pub fn parse_response<B>(reader: BaseStream, request: &PreparedRequest<B>) -> Re
     headers.remove(TRANSFER_ENCODING);
 
     Ok(Response {
+        url: url.clone(),
```
Comment (sbstp) [inline src/parsing/response.rs:89]: wouldn't `request.url()` be the same and avoid adding a param?

Comment (sbstp) [reply src/parsing/response.rs:89]: nevermind, no, if there's a redirect it will be the wrong url

Comment (baskerville) [reply src/parsing/response.rs:89]: Yes, I'll just need to update `self.url` in `send`'s loop.

Comment (sbstp) [reply src/parsing/response.rs:89]: I think that what you are currently doing is better. I like to think of the PreparedRequest as immutable.

**Comment by sbstp** [comment]
Thanks

**Comment by baskerville** [comment]
If #138 gets merged, would you mind making a new release?

**Pull request merged**